src/java.base/share/classes/sun/security/ssl/SSLAlgorithmConstraints.java

```diff
@@ -69,6 +69,13 @@ final class SSLAlgorithmConstraints implements AlgorithmConstraints {
         this.enabledX509DisabledAlgConstraints = true;
     }
 
+    static AlgorithmConstraints wrap(AlgorithmConstraints userSpecifiedConstraints) {
```

Comment:
I may update all of the constructors so that the accumulation of the reference of userSpecifiedConstraints could be avoid further.

```
-       this.userSpecifiedConstraints = userSpecifiedConstraints;
+       this.userSpecifiedConstraints = userSpecifiedConstraints == DEFAULT ?
+               null : userSpecifiedConstraints;

```

Similar update could be placed in the getUserSpecifiedConstraints() implementation.

Reply:
Thanks @XueleiFan for the review!
If we do that, this will result in a behavior change for cases where `enabledX509DisabledAlgConstraints` = false; is that okay? Or should we set `enabledX509DisabledAlgConstraints` = true if `userSpecifiedConstraints == DEFAULT`?

Reply:
updated the patch. Let me know if that's what you had in mind.